vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,164 @@
+        yamltext, mdtext = split_markdown_front_matter(lines.read())
+        data = yaml.safe_load(yamltext)
+        data["mfsa_id"] = mfsa_id
+
+        fixed_package_urls = self.get_package_urls(data.get("fixed_in"))
+        references = self.get_yml_references(data)
+        cves = re.findall(r"CVE-\d+-\d+", yamltext + mdtext, re.IGNORECASE)
+        for cve in cves:
+            references.append(
+                Reference(
+                    reference_id=cve, url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve}"
+                )
+            )
+
+        description = self.html_get_p_under_h3(markdown(mdtext), "description")
+
+        return [
+            Advisory(
+                summary=description,
+                vulnerability_id="",
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        ]
+
+    def html_get_p_under_h3(self, html, h3: str):
+        soup = BeautifulSoup(html, features="lxml")
+        h3tag = soup.find("h3", text=lambda txt: txt.lower() == h3)
+        p = ""
+        if h3tag:
+            for tag in h3tag.next_siblings:
+                if tag.name:
+                    if tag.name != "p":
+                        break
+                    p += tag.get_text()
+        return p
+
+    def mfsa_id_from_filename(self, filename):
+        match = MFSA_FILENAME_RE.search(filename)
+        if match:
+            return "mfsa" + match.group(1)
+
+        return None
+
+    def get_package_urls(self, pkgs: List[str]) -> List[PackageURL]:
+        package_urls = [
+            PackageURL(
+                type="mozilla",
+                # pkg is of the form "Firefox ESR 1.21" or "Thunderbird 2.21"
+                name=pkg.rsplit(None, 1)[0],
+                version=pkg.rsplit(None, 1)[1],
+            )
+            for pkg in pkgs
+        ]
+        return package_urls
+
+    def get_yml_references(self, data: any) -> List[Reference]:
+        """
+        Returns a list of references
+        Currently only considers the given mfsa as a reference
+        """
+        # FIXME: Needs improvement
+        # Should we add 'bugs' section in references too?
+        # Should we add 'impact'/severity of CVE in references too?
+        # If yes, then fix alpine_linux importer as well
+        # Otherwise, do we need severity field for adversary as well?
+
+        # FIXME: Write a helper for cvssv3.1_qr severity detection ?
+        severities = ["critical", "low", "high", "medium", "none"]
+        severity = [{severity in data.get("impact"): severity} for severity in severities][0].get(
+            True
+        )
+
+        return [
+            Reference(
+                reference_id=data["mfsa_id"],
+                url="https://www.mozilla.org/en-US/security/advisories/{}".format(data["mfsa_id"]),
+                severities=[VulnerabilitySeverity(scoring_systems["cvssv3.1_qr"], severity)],
```

Comment:
It's not a cvssv3.1_qr , create a system for these.

Reply:
Done. See #415